### DISCUSSION TITLE
Allow prop for custom error component

### DISCUSSION THREAD

**Discussion by onlinewithryan** [gilbarbara/react-spotify-web-playback#110] [Ideas]
I was planning on opening a pull request for this feature but opening an issue to save everyone some time. 

I want to add a prop to allow the caller to provide a react component to render (or not render) errors. 

For example:

```   
 if (status === STATUS.ERROR) {
      const {CustomErrorComponent} = this.props;
      output = CustomErrorComponent ? <CustomErrorComponent error={error} /> : (
        <ErrorMessage styles={this.styles}>
          {name}: {error}
        </ErrorMessage>
      );
    }
```

Today we can get the most recent error through `this.props.callback` but it does not allow us to render custom error handling without unrendering the web player, which will prevent any retries and any other updates from flowing down.

**Comment by gilbarbara** [comment]
Hey @onlinewithryan 

> I've moved this to discussions since it's not a bug.

I've been thinking about error handling for the UI update I'm planning (#41), displaying non-fatal errors without destroying the UI, and how to recover from them.

I don't think just hiding the error would be helpful since the UI would be replaced by this component anyway.
Also, the error `name` is important because the message can differ in some scenarios.

Maybe a function to reset the state if it's possible to recover? Can you elaborate on the issues you are having?

Anyway, I don't like adding props without first thinking about them to avoid an a-prop-calypse. :)

**Reply by onlinewithryan** [reply]
I guess I have seen some weird behavior where an error my show, but then another async response was in flight. That response then put the player in a valid state, these have been flakey and likely cause by networking issues. I was looking to not show anything to the end user on these kind of errors since the player loading just adds to the experience but is not required.

You are right that this kind of UI error handling can be dealt with from outside the component. 
```
      if (error === 'authentication_error' && this.hasNewToken) {
        this.hasNewToken = false;
        this.initializePlayer();
      }
```
I'm fine with closing this as not important
